docfx_yaml/extension.py

```diff
@@ -1418,13 +1430,21 @@ def disambiguate_toc_name(pkg_toc_yaml):
         if module_name not in name_entries:
             name_entries[module_name] = {}
 
+        prev_part = ""
         # Split the name and mark all duplicates.
         # There will be at least 1 unique identifer for each name.
         for part in module['uidname'].split("."):
-            if part not in name_entries[module_name]:
+            if part not in name_entries[module_name] or (
+                # If the name appears in a row, for example:
+                #   google.cloud.run_v2.services
+                #   google.cloud.run_v2.services.services
+                # then ignore this entry as this is clear from the left-nav layout.
+                part == prev_part
+            ):
                 name_entries[module_name][part] = 1
```

Comment:
I think this will work exclusively when the duplicate identifier is at the end. Like in your example, for `google.cloud.run_v2.services.services` it will see that `prev_part = 'services'` and reset the entry count to 1. However, if it was like `google.cloud.run_v2.services.services.jobs` then the entry count would go up to 2, bringing  it back into the TOC (if I'm not mistaken). Is that case a concern?

Reply:
The mapping would be like this:
```
'run_v2' = 1
'services' = 1
'jobs' = 1
```
in order for this to be a duplicate, we'd need another UID that ends with `jobs`. For example:

```
google.cloud.run_v2.services.services.jobs
google.cloud.run_v2.types.jobs
```
in this case, we can disambiguate with `types.jobs` and `services.jobs`.

Another example could be:

```
google.cloud.run_v1.services.services.jobs
google.cloud.run_v2.services.jobs
```
In this case we'd have `run_v1.jobs` and `run_v2.jobs` as disambiguators. `services` might also get disambiguated with `run_v1` or v2 prefixes, but in all cases this works. I _don't_ think I've left any other cases out. Sorry that this is confusing! :S

Reply:
Got it -- in that case LGTM